Show page banner for anonymous and unenrolled users

andy-armstrong opened the pull request:
## [LEARNER-1697](https://openedx.atlassian.net/browse/LEARNER-1697)

### Description

This change introduces two new messages in the page banner for course pages that can be viewed by unenrolled/anonymous users. For now, this means the message is shown on the unified "Course" tab and on static pages. I chose not to implement this on the current course info page since that already has a message, and the page will be going away.

1. Anonymous user

![image](https://user-images.githubusercontent.com/5985072/28241992-44ccb19a-696e-11e7-86f1-71cc2341995d.png)

2. Unenrolled user

![image](https://user-images.githubusercontent.com/5985072/28242001-5f6c2f6c-696e-11e7-865b-07dd1644a50e.png)

### Sandbox
- [ ] Try these two URLs when logged out or when using honor@example.com:

https://andy-armstrong.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/course/

https://andy-armstrong.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/ebf94dcfafbc4b61a5f2efafe010024a/

### Testing
- [ ] Unit, integration, acceptance tests as appropriate
- [ ] Analytics
- [ ] Performance
- [ ] Database migrations are backwards-compatible
- [ ] safecommit violation code review process

Front-end changes:
- [ ] i18n
- [ ] Accessibility (Check for a11y violations, ensure keyboard accessible, screen reader testing as appropriate)
- [ ] RTL

### Reviewers

FYI: @marcotuts @lizcohen @HarryRein @robrap @dianakhuang @catong 
 
### Post-review
- [ ] Rebase and squash commits

andy-armstrong commented:
jenkins run python

lizcohen commented:
@andy-armstrong please add underline to links

robrap commented on lms/djangoapps/courseware/views/views.py:
```diff
@@ -440,6 +439,52 @@ def get(self, request, course_id, tab_type, **kwargs):
+                request,
+                Text(_("To see course content, {sign_in_link} or {register_link}.")).format(
+                    sign_in_link=HTML('<a href="/login?next={current_url}">{sign_in_label}</a>').format(
+                        sign_in_label=_("Sign in"),
```
@andy-armstrong A reminder to change this to lower case `sign in`.

andy-armstrong replied:
Done

robrap commented on lms/djangoapps/courseware/views/views.py:
```diff
@@ -440,6 +439,52 @@ def get(self, request, course_id, tab_type, **kwargs):
+                Text(_('You must be enrolled in the course to see course content. {enroll_button}')).format(
+                    enroll_button=HTML('<a href="{url_to_enroll}">{enroll_button_label}</a>').format(
+                        url_to_enroll=CourseTabView.url_to_enroll(course_key),
+                        enroll_button_label=_("Enroll Now"),
```
@andy-armstrong @catong: The old message on the info page read: "Enroll now!".  I prefer the lowercase 'n'.  I also think we need punctuation (period or exclamation point).

I made the Performance course use the info page and unenrolled the honor student, so you can see this message here when logged in as honor:
https://andy-armstrong.sandbox.edx.org/courses/course-v1:DemoX+PERF101+course/

catong replied:
@robrap Agreed. This should be written and punctuated as if it's sentence text: "Enroll now." (plus link underlining).

andy-armstrong replied:
@robrap I thought I'd taken this from here:

https://openedx.atlassian.net/wiki/pages/viewpage.action?pageId=160609943

but it looks like it should have been a lower case 'n'.

![image](https://user-images.githubusercontent.com/5985072/28283776-d072e716-6afc-11e7-9fa6-f34a9ed0ff62.png)

 I think we need to decide to either make all of these messages use full sentences, or we should use buttons. @catong @lizcohen, what do you recommend?

andy-armstrong replied:
@catong I missed your comment! I'll go with your proposal

robrap commented:
@andy-armstrong I imagine this might be too much to fix (I don't look), but just as an FYI:
```
Linting lms/templates/main.html:
lms/templates/main.html: mako-missing-default

1 violations total
```

andy-armstrong commented on lms/djangoapps/courseware/views/views.py:
```diff
@@ -440,6 +439,52 @@ def get(self, request, course_id, tab_type, **kwargs):
+        """
+        Returns the URL to use to enroll in the specified course.
+        """
+        # TODO: LEARNER-1697: Transition handling of enroll links in new Course Home.
```
Good point. Done

andy-armstrong commented on lms/djangoapps/courseware/views/views.py:
```diff
@@ -440,6 +439,52 @@ def get(self, request, course_id, tab_type, **kwargs):
+    @staticmethod
+    def show_user_access_warnings(request, course_key):
+        """
+        Show messages to the user if they have limited access.
```
I agree. I like your name, so I've switched it.

andy-armstrong commented:
jenkins run python

andy-armstrong commented:
@robrap @lizcohen @catong I've addressed your feedback and updated my sandbox. Please re-review when you have a chance.

robrap commented:
Nice job @andy-armstrong. 

I don't see a response to my 404 question from my summary comment from the last review.

Also, I've got one new minor comment that I need to clean up in the morning at my computer.

andy-armstrong commented:
@robrap Sorry I missed your comment about the 404 page. I've fixed it by moving the message registration to be after the URL validation check. I've also removed the TODO since we're going to fix enrollment this sprint anyway. I'm going to go ahead and merge if this gets a clean build.

robrap commented:
Great.  Nice job.
